drivers/serial/uart_ns16550.c

```diff
@@ -729,6 +809,10 @@ static int uart_ns16550_init(const struct device *dev)
 
 		device_map(DEVICE_MMIO_RAM_PTR(dev), mbar.phys_addr, mbar.size,
 			   K_MEM_CACHE_NONE);
+#if defined(CONFIG_UART_ASYNC_API)
+		pcie_set_cmd(dev_cfg->pcie->bdf, PCIE_CONF_CMDSTAT_MASTER, true);
+		data->phys_addr = mbar.phys_addr;
+#endif
```

Comment:
This block is specific to LPSS (since its under PCIe) and not generic. Please do #ifdef appropriately.

Reply:
This is specific to any UART instance under PCIe bus with ASYNC support with DMA, not related to LPSS.